src/os_unix.c

```diff
@@ -8256,3 +8262,129 @@ xsmp_close(void)
     }
 }
 #endif // USE_XSMP
+
+#ifdef FEAT_RELTIME
+static struct itimerval prev_interval;
+static struct sigaction prev_sigaction;
+static int timeout_flag         = FALSE;
```

Comment:
is it risky that this is global? i.e. if start_timeout is called-reentrantly? 

is it possible for the regex code to be re-entrant? A lot of vim code has to handle this sort of thing.